lib/server-parser.js

```diff
@@ -0,0 +1,377 @@
+    help: 'Add long stack traces to log entries. Recommended for debugging only.',
+  }],
+
+  [['--webkit-debug-proxy-port'], {
+    defaultValue: 27753,
+    dest: 'webkitDebugProxyPort',
+    required: false,
+    type: 'int',
+    example: '27753',
+    help: '(IOS-only) Local port used for communication with ios-webkit-debug-proxy'
+  }],
+
+  [['--webdriveragent-port'], {
+    defaultValue: 8100,
+    dest: 'wdaLocalPort',
+    required: false,
+    type: 'int',
+    example: '8100',
+    help: '(IOS-only, XCUITest-only) Local port used for communication with WebDriverAgent'
+  }],
+
+  [['-dc', '--default-capabilities'], {
+    dest: 'defaultCapabilities',
+    defaultValue: {},
+    type: parseDefaultCaps,
+    required: false,
+    example: '[ \'{"app": "myapp.app", "deviceName": "iPhone Simulator"}\' ' +
+             '| /path/to/caps.json ]',
+    help: 'Set the default desired capabilities, which will be set on each ' +
+          'session unless overridden by received capabilities.'
+  }],
+
+  [['--relaxed-security'], {
+    defaultValue: false,
+    dest: 'relaxedSecurityEnabled',
+    action: 'storeTrue',
+    required: false,
+    help: 'Disable additional security checks, so it is possible to use some advanced features, provided ' +
+          'by drivers supporting this option. Only enable it if all the ' +
+          'clients are in the trusted network and it\'s not the case if a client could potentially ' +
+          'break out of the session sandbox. Specific features can be overridden by ' +
+          'using the --deny-insecure flag',
+    nargs: 0
+  }],
+
+  [['--allow-insecure'], {
+    dest: 'allowInsecure',
+    defaultValue: [],
+    type: parseSecurityFeatures,
+    required: false,
+    example: 'execute_driver_script,adb_shell',
+    help: 'Set which insecure features are allowed to run in this server\'s sessions. ' +
+          'Features are defined on a driver level; see documentation for more details. ' +
+          'This should be either a comma-separated list of feature names, or a path to ' +
+          'a file where each feature name is on a line. Note that features defined via ' +
+          '--deny-insecure will be disabled, even if also listed here.',
+  }],
+
+  [['--deny-insecure'], {
+    dest: 'denyInsecure',
+    defaultValue: [],
+    type: parseSecurityFeatures,
+    required: false,
+    example: 'execute_driver_script,adb_shell',
+    help: 'Set which insecure features are not allowed to run in this server\'s sessions. ' +
+          'Features are defined on a driver level; see documentation for more details. ' +
+          'This should be either a comma-separated list of feature names, or a path to ' +
+          'a file where each feature name is on a line. Features listed here will not be ' +
+          'enabled even if also listed in --allow-insecure, and even if --relaxed-security ' +
+          'is turned on.',
+  }],
+];
+
+function parseSecurityFeatures (features) {
```

Comment:
There is a potential issue that some third party drivers could mimic the same feature names. Perhaps, it would make sense to prefix each feature name with the corresponding automation name it affects (e.g. UiAutomator2:adb_shell)

Reply:
this is a good point, however given that we can't prevent the third party drivers from just `rm -rf /` anyway, there doesn't seem to be much use in trying to create a restricted system. to even use a 3rd party driver you have to trust it totally. authors of 3rd party drivers can opt into appium's security model, just the way we do as 1st party authors. if they want to do something nefarious, they don't need to worry about breaking our security model, they can just do it.